flink-table/flink-table-common/src/main/java/org/apache/flink/table/descriptors/Rowtime.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.descriptors;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.table.sources.tsextractors.TimestampExtractor;
+import org.apache.flink.table.sources.wmstrategies.WatermarkStrategy;
+
+import java.util.Map;
+
+/**
+ * Rowtime descriptor for describing an event time attribute in the schema.
+ */
+@PublicEvolving
+public class Rowtime implements Descriptor {
+
+	private final DescriptorProperties internalProperties = new DescriptorProperties(true);
+
+	public static final String ROWTIME = "rowtime";
```

Comment:
These static fields should rather go into a `RowtimeValidator` class as they mess up the API class otherwise.

Reply:
Make sense. As we can not port `RowtimeValidator` into the common module now, how about extract these properties into a new property class? This not only makes API more clear but also makes these variables more well arranged. What do you think?

Reply:
I'm also fine with keeping the fields there until we ported `RowtimeValidator` but with a TODO comment to move them later.

Reply:
Ok, sounds good. I will add a TODO comment. Thank you.